This platform requires CONFIG_GRUB_SERIAL be set

### Describe the bug

Download package
……

make[2] -C feeds/luci/applications/luci-app-firewall download

make[2] -C package/boot/uboot-armsr download

make[2] -C feeds/luci/applications/luci-app-frps download

make[2] -C feeds/luci/applications/luci-app-nft-qos download

ERROR: package/base-files failed to build.

make package/download: build failed. Please re-run make with -j1 V=s or V=sc for a higher verbosity level to see what's going on

make[1] target/download

make[2] -C target/linux download


=========================================================

Compile the fireware

……
make[3]: Leaving directory '/workdir/openwrt/package/system/usign'

time: package/system/usign/compile#0.19#0.08#0.23

make[3]: Entering directory '/workdir/openwrt/package/base-files'

/workdir/openwrt/target/linux/armsr/base-files.mk:3: *** This platform requires CONFIG_GRUB_SERIAL be set!. Stop.

make[3]: Leaving directory '/workdir/openwrt/package/base-files'

time: package/base-files/compile#0.66#0.05#0.72

ERROR: package/base-files failed to build.

make[2]: *** [package/Makefile:128: package/base-files/compile] Error 1

make[2]: Leaving directory '/workdir/openwrt'

make[1]: *** [package/Makefile:122: /workdir/openwrt/staging_dir/target-aarch64_generic_musl/stamp/.package_compile] Error 2

make[1]: Leaving directory '/workdir/openwrt'

make: *** [/workdir/openwrt/include/toplevel.mk:232: world] Error 2

Error: Process completed with exit code 2.

### OpenWrt version

OpenWrt version VERSION_INFO:  useVersionInfo: Author: Daniel Golle<br/>date: 2023-11-25<br/>commit: mediatek: remove DTS property added by mistake<br/>commit hash: 1b70025ad1f5d8dd93e8415de7817d7975d4a8d2  DATE: 2023-11-25

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

armsr/armv8

### Device

Phicomm N1

### Image kind

Self-built image

### Steps to reproduce

_No response_

### Actual behaviour

_No response_

### Expected behaviour

_No response_

### Additional info

_No response_

### Diffconfig

_No response_

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

--- a/config/Config-images.in
+++ b/config/Config-images.in
@@ -218,11 +218,6 @@ menu "Target Images"
 		depends on GRUB_IMAGES || GRUB_EFI_IMAGES
 		default y
 
-	config GRUB_SERIAL
-		string "Serial port device"
-		depends on GRUB_IMAGES || GRUB_EFI_IMAGES
-		default "ttyS0"
-
 	config GRUB_BAUDRATE
 		int "Serial port baud rate"
 		depends on GRUB_IMAGES || GRUB_EFI_IMAGES
@@ -231,7 +226,8 @@ menu "Target Images"
 
 	config GRUB_FLOWCONTROL
 		bool "Use RTE/CTS on serial console"
-		depends on GRUB_SERIAL != ""
+		depends on GRUB_IMAGES || GRUB_EFI_IMAGES
+		depends on TARGET_SERIAL != ""
 
 	config GRUB_BOOTOPTS
 		string "Extra kernel boot options"
@@ -276,6 +272,11 @@ menu "Target Images"
 		depends on TARGET_x86
 		depends on GRUB_IMAGES || GRUB_EFI_IMAGES
 		select PACKAGE_kmod-e1000
+
+	config TARGET_SERIAL
+		string "Serial port device"
+		depends on TARGET_x86 || TARGET_armsr
+		default "ttyS0"
 
 	config TARGET_IMAGES_GZIP
 		bool "GZip images"

--- a/target/linux/armsr/base-files.mk
+++ b/target/linux/armsr/base-files.mk
@@ -1,6 +1,6 @@
-GRUB_SERIAL:=$(call qstrip,$(CONFIG_GRUB_SERIAL))
+GRUB_SERIAL:=$(call qstrip,$(CONFIG_TARGET_SERIAL))
 ifeq ($(GRUB_SERIAL),)
-$(error This platform requires CONFIG_GRUB_SERIAL be set!)
+$(error This platform requires CONFIG_TARGET_SERIAL be set!)
 endif
 
 define Package/base-files/install-target

--- a/target/linux/armsr/image/Makefile
+++ b/target/linux/armsr/image/Makefile
@@ -15,7 +15,7 @@ ifneq ($(CONFIG_GRUB_CONSOLE),)
   GRUB_TERMINALS += console
 endif
 
-GRUB_SERIAL:=$(call qstrip,$(CONFIG_GRUB_SERIAL))
+GRUB_SERIAL:=$(call qstrip,$(CONFIG_TARGET_SERIAL))
 
 GRUB_SERIAL_CONFIG := serial --unit=0 --speed=$(CONFIG_GRUB_BAUDRATE) --word=8 --parity=no --stop=1 --rtscts=$(if $(CONFIG_GRUB_FLOWCONTROL),on,off)
 GRUB_TERMINALS += serial

--- a/target/linux/x86/base-files.mk
+++ b/target/linux/x86/base-files.mk
@@ -1,6 +1,6 @@
-GRUB_SERIAL:=$(call qstrip,$(CONFIG_GRUB_SERIAL))
+GRUB_SERIAL:=$(call qstrip,$(CONFIG_TARGET_SERIAL))
 ifeq ($(GRUB_SERIAL),)
-$(error This platform requires CONFIG_GRUB_SERIAL be set!)
+$(error This platform requires CONFIG_TARGET_SERIAL be set!)
 endif
 
 define Package/base-files/install-target

--- a/target/linux/x86/image/Makefile
+++ b/target/linux/x86/image/Makefile
@@ -16,7 +16,7 @@ ifneq ($(CONFIG_GRUB_CONSOLE),)
   GRUB_TERMINALS += console
 endif
 
-GRUB_SERIAL:=$(call qstrip,$(CONFIG_GRUB_SERIAL))
+GRUB_SERIAL:=$(call qstrip,$(CONFIG_TARGET_SERIAL))
 
 GRUB_CONSOLE_CMDLINE += console=$(GRUB_SERIAL),$(CONFIG_GRUB_BAUDRATE)n8$(if $(CONFIG_GRUB_FLOWCONTROL),r,)
 GRUB_SERIAL_CONFIG := serial --unit=0 --speed=$(CONFIG_GRUB_BAUDRATE) --word=8 --parity=no --stop=1 --rtscts=$(if $(CONFIG_GRUB_FLOWCONTROL),on,off)

